src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -974,6 +974,49 @@ class IpexCommunicationService extends AgentService {
       status: CredentialStatus.PENDING,
     };
   }
+
+  async getLinkedGroupFromIpexGrant(id: string) {
+    const grantNoteRecord = await this.notificationStorage.findById(id);
+
+    if (!grantNoteRecord) {
+      throw new Error(
+        `${IpexCommunicationService.NOTIFICATION_NOT_FOUND} ${id}`
+      );
+    }
+
+    const membersJoined: string[] = [];
+    const linkedGroupRequest = grantNoteRecord.linkedGroupRequests;
+    const exchange = await this.props.signifyClient
+      .exchanges()
+      .get(grantNoteRecord.a.d as string);
+
+    const credentialSaid = exchange.exn.e.acdc.d;
+
+    if (Object.keys(linkedGroupRequest).length) {
+      const saids = linkedGroupRequest[credentialSaid].saids;
+
+      for (const key in saids) {
+        if (key in saids) {
+          const memberDetails = saids[key];
+
+          if (memberDetails.length > 0 && memberDetails[0].length > 0) {
+            const member = memberDetails[0][0];
+            membersJoined.push(member);
```

Comment:
Because there can be multiple admit SAIDs, we can have duplicate members. We want to remove duplicates. Can we de-duplicate after with a set before returning?